packages/manager/modules/pci/src/projects/project/public-gateways/public-gateways.service.js

```diff
@@ -1,22 +1,111 @@
-export default class publigGatewaysServiceClass {
-  static getIcebergHeaders() {
-    return {
-      headers: {
-        'X-Pagination-Mode': 'CachedObjectList-Pages',
-        'X-Pagination-Size': 50000,
-        Pragma: 'no-cache',
-      },
-    };
-  }
+import { GATEWAY_DEFAULT_REGION } from './public-gateways.constants';
 
+export default class publicGatewaysServiceClass {
   /* @ngInject */
-  constructor(OvhApiCloudProjectNetworkPrivate, $http) {
+  constructor(
+    OvhApiCloudProjectNetworkPrivate,
+    $http,
+    $q,
+    OvhApiCloudProject,
+    PciProjectRegions,
+    ovhManagerRegionService,
+    Poller,
+    OvhApiCloudProjectNetworkPrivateSubnet,
+  ) {
+    this.$q = $q;
     this.$http = $http;
+    this.OvhApiCloudProjectNetworkPrivate = OvhApiCloudProjectNetworkPrivate;
+    this.OvhApiCloudProject = OvhApiCloudProject;
+    this.PciProjectRegions = PciProjectRegions;
+    this.ovhManagerRegionService = ovhManagerRegionService;
+    this.Poller = Poller;
+    this.OvhApiCloudProjectNetworkPrivateSubnet = OvhApiCloudProjectNetworkPrivateSubnet;
+    this.region = GATEWAY_DEFAULT_REGION;
   }
 
-  getPublicGateways(serviceName) {
+  getGatwayCatalog(params) {
     return this.$http
-      .get(`/cloud/project/${serviceName}/gateway`)
+      .get(`/order/catalog/public/cloud`, {
+        params,
+      })
       .then(({ data }) => data);
   }
+
+  getGateways(serviceName, regionName = this.region) {
+    return this.$http
+      .get(`/cloud/project/${serviceName}/region/${regionName}/gateway`)
+      .then((gateways) =>
+        gateways.data.map((gateway) => {
+          return {
+            ...gateway,
+            region: this.region,
+            formattedIps: gateway.externalInformation.ips
+              .map((e) => e.ip)
+              .join(', '),
+          };
+        }),
+      )
+      .catch((err) => this.$q.reject(err));
+  }
+
+  addGateway(serviceName, region, gateway) {
+    return this.$http
+      .post(`/cloud/project/${serviceName}/region/${region}/gateway `, gateway)
+      .then(({ data }) => data)
+      .catch((err) => this.$q.reject(err));
```

Comment:
I think implement `catch` into service file is useless.
You have to implement them where the request is implement.

Reply:
yes, make sense and updated as suggested.